// @hubspot/api-client/lib/src/discovery/cms/CmsDiscovery.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const initObject_1 = require("../../services/initObject");
const BaseDiscovery_1 = __importDefault(require("../BaseDiscovery"));
class CmsDiscovery extends BaseDiscovery_1.default {
    get auditLogs() {
        if (!this._auditLogs) {
            this._auditLogs = (0, initObject_1.initObject)('cms/audit_logs/AuditLogsDiscovery', this.config);
        }
        return this._auditLogs;
    }
    get blogs() {
        if (!this._blogs) {
            this._blogs = (0, initObject_1.initObject)('cms/blogs/BlogsDiscovery', this.config);
        }
        return this._blogs;
    }
    get domains() {
        if (!this._domains) {
            this._domains = (0, initObject_1.initObject)('cms/domains/DomainsDiscovery', this.config);
        }
        return this._domains;
    }
    get hubdb() {
        if (!this._hubdb) {
            this._hubdb = (0, initObject_1.initObject)('cms/hubdb/HubdbDiscovery', this.config);
        }
        return this._hubdb;
    }
    get performance() {
        if (!this._performance) {
            this._performance = (0, initObject_1.initObject)('cms/performance/PerformanceDiscovery', this.config);
        }
        return this._performance;
    }
    get siteSearch() {
        if (!this._siteSearch) {
            this._siteSearch = (0, initObject_1.initObject)('cms/site_search/SiteSearchDiscovery', this.config);
        }
        return this._siteSearch;
    }
    get sourceCode() {
        if (!this._sourceCode) {
            this._sourceCode = (0, initObject_1.initObject)('cms/source_code/SourceCodeDiscovery', this.config);
        }
        return this._sourceCode;
    }
    get urlRedirects() {
        if (!this._urlRedirects) {
            this._urlRedirects = (0, initObject_1.initObject)('cms/url_redirects/UrlRedirectsDiscovery', this.config);
        }
        return this._urlRedirects;
    }
}
exports.default = CmsDiscovery;
